internal/scheduler/scheduler.go

```diff
@@ -167,7 +170,18 @@ func (s *Scheduler) Run(ctx context.Context) error {
 				logging.WithStacktrace(log, err).Error("scheduling cycle failure")
 				leaderToken = InvalidLeaderToken()
 			}
-			log.Infof("scheduling cycle completed in %s", s.clock.Since(start))
+
+			cycleTime := s.clock.Since(start)
+
+			if shouldSchedule && leaderToken.leader {
+				// Only the leader token does real scheduling rounds.
+				s.metrics.ReportScheduleCycleTime(float64(cycleTime.Milliseconds()))
```

Comment:
Better to give a `time.Duration` imo since that captures the unit.

Reply:
histogram doesn't handle time.Duration well.

Reply:
You could do the conversion in (metrics *SchedulerMetrics) ReportScheduleCycleTime tho.